### PR TITLE
http3: implement FlushError for the response writer

### PR DESCRIPTION
**Pull request by WeidiDeng**
Move all of the logic to the new `FlushError` function and let `Flush` call it.

### CODE REVIEW

File: http3/response_writer.go [hunk 1]
```diff
@@ -114,10 +114,16 @@ func (w *responseWriter) Write(p []byte) (int, error) {
 	return w.bufferedStr.Write(p)
 }
 
-func (w *responseWriter) Flush() {
+func (w *responseWriter) FlushError() error {
 	if err := w.bufferedStr.Flush(); err != nil {
 		w.logger.Errorf("could not flush to stream: %s", err.Error())
```
Comment (marten-seemann) [inline http3/response_writer.go:119]: No need to log if you're returning the error.

File: http3/response_writer.go [hunk 2]
```diff
@@ -114,10 +114,16 @@ func (w *responseWriter) Write(p []byte) (int, error) {
...
+		return err
 	}
+	return nil
+}
+
+func (w *responseWriter) Flush() {
+	w.FlushError()
```
Comment (marten-seemann) [inline http3/response_writer.go:126]: Log here if you're ignoring the error.

**Review by marten-seemann — COMMENTED** [review]
`Flush` is an implementation of `http.Flusher`. I don't see any corresponding `http.FlusherError` interface.

What is the motivation behind this change?

**Comment by WeidiDeng** [comment]
@marten-seemann `FlushError` is used by [ResponseController](https://pkg.go.dev/net/http#NewResponseController), like setdeadline related methods.

**Review by marten-seemann — APPROVED** [review]
Thank you!

**Pull request merged**